src/sqlfluff/dialects/dialect_mysql.py

```diff
@@ -531,6 +534,18 @@ class InsertStatementSegment(BaseSegment):
     )
 
 
+class IndentedFromExpressionSegment(FromExpressionSegment):
+    """An indented from expression segment."""
+
+    type = "indented_from_expression"
+
+    match_grammar = Sequence(
+        Indent,
+        Ref("FromExpressionSegment"),
+        Dedent,
+    )
+
+
```

Comment:
I don't see the need for this class, and the extra level it creates in the parse tree. I agree it is used twice, but think cleaner to just `Indent`/`Dedent` directly in both those occasions - like the other dialects do.

This would also reduce some of the changes in these PR (including most of the mysql YAML changes?).

```suggestion
```